lighthouse-viewer/app/src/drag-and-drop.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * Manages drag and drop file input for the page.
+ */
+class DragAndDrop {
+  /**
+   * @param {function(!File)} fileHandlerCallback Invoked when the user chooses a new file.
+   */
+  constructor(fileHandlerCallback) {
+    this._dropZone = document.querySelector('.drop_zone');
+    this._fileHandlerCallback = fileHandlerCallback;
+    this._dragging = false;
+
+    this._addListeners();
+  }
+
+  _addListeners() {
+    // The mouseleave event is more reliable than dragleave when the user drops
+    // the file outside the window.
+    document.addEventListener('mouseleave', _ => {
+      if (!this._dragging) {
+        return;
+      }
+      this._resetDraggingUI();
+    });
+
+    document.addEventListener('dragover', e => {
+      e.stopPropagation();
+      e.preventDefault();
```

Comment:
This dude told me to do it: https://www.html5rocks.com/en/tutorials/dnd/basics/

TBH, I don't think we need them anymore in a `dragover` handler. Back in the day, there were browser quirks and I noted that `e.preventDefault()` was needed to even start a drag. But we're also dragging files (as opposed to links or something in-page) and at the document level. [MDN mentions](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API#Define_a_drop_zone) `e.preventDefault` in `dragover`, but I betcha that's not accurate.

Anyways, doesn't hurt to be explicit the method is handing the event.